lib/wallet/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -66,269 +57,253 @@ import Cardano.Wallet.Primitive.Types
+        it "Checkpoint"
+            $ checkCoverage
+            $ prop_sequentialPut
+                testOnLayer
+                (\DBLayer {..} a0 -> mapExceptT atomically . putCheckpoint a0)
+                (\DBLayer {..} -> atomically . readCheckpoint)
+                lastMay
+                . sortOn (currentTip . unShowFmt)
+        it "Wallet Metadata"
+            $ checkCoverage
+            $ prop_sequentialPut
+                testOnLayer
+                (\DBLayer {..} a0 -> mapExceptT atomically . putWalletMeta a0)
+                (\DBLayer {..} -> atomically . fmap (fmap fst) . readWalletMeta)
+                lastMay
+        it "Tx History"
+            $ checkCoverage
+            $ prop_sequentialPut
+                testOnLayer
                 putTxHistory_
                 readTxHistory_
-                (length . sortedUnions)
-        it "Private Key" $
-            checkCoverage . prop_parallelPut
-                (\DBLayer{..} a0 -> mapExceptT atomically . putPrivateKey a0)
-                (\DBLayer{..} -> atomically . readPrivateKey)
-                (length . lrp @Maybe)
+                ( let sort' =
+                        GenTxHistory
+                            . filterTxHistory Nothing Descending wholeRange
+                            . unGenTxHistory
+                  in  Identity . sort' . fold
+                )
+        it "Private Key"
+            $ checkCoverage
+            $ prop_sequentialPut
+                testOnLayer
+                (\DBLayer {..} a0 -> mapExceptT atomically . putPrivateKey a0)
+                (\DBLayer {..} -> atomically . readPrivateKey)
+                lastMay
 
     describe "rollback" $ do
-        it "Can rollback to any arbitrary known checkpoint"
-            (property . prop_rollbackCheckpoint)
-        it "Correctly re-construct tx history on rollbacks"
-            (checkCoverage . prop_rollbackTxHistory)
+        it
+            "Can rollback to any arbitrary known checkpoint"
+            (property $ prop_rollbackCheckpoint withFreshDB)
+        it
+            "Correctly re-construct tx history on rollbacks"
+            (checkCoverage $ prop_rollbackTxHistory withFreshDB)
 
 -- | Wrap the result of 'readTransactions' in an arbitrary identity Applicative
 readTxHistory_
     :: Functor m
     => DBLayer m s ShelleyKey
     -> WalletId
     -> m (Identity GenTxHistory)
-readTxHistory_ DBLayer{..} wid =
+readTxHistory_ DBLayer {..} wid =
     (Identity . GenTxHistory . fmap toTxHistory)
         <$> atomically
             (readTransactions wid Nothing Descending wholeRange Nothing Nothing)
 
 putTxHistory_
-    :: DBLayer m s ShelleyKey
+    :: Functor m
+    => DBLayer m s ShelleyKey
     -> WalletId
     -> GenTxHistory
-    -> ExceptT ErrNoSuchWallet m ()
-putTxHistory_ DBLayer{..} wid =
-    mapExceptT atomically . putTxHistory wid . unGenTxHistory
+    -> ExceptT ErrWalletNotInitialized m ()
+putTxHistory_ DBLayer {..} wid =
```

Comment:
```suggestion
putTxHistory_ DBLayer{..} wid =
```

Formatting: Avoid space before record bracket in order to avoid confusing the `{…}` with a standalone function argument.

Reply:
fixed fourmolu.yaml , good they support it :)